cmd/tools/rest/href_builder_test.go

```diff
@@ -0,0 +1,101 @@
+package rest
+
+import (
+	"github.com/google/go-cmp/cmp"
+	"strconv"
+	"testing"
+)
+
+func TestBuild(t *testing.T) {
+	testQuery := "storage/volumes"
+	testFields := []string{"name", "svm"}
+	testHiddenFields := []string{"statistics"}
+	testFilter := []string{""}
+	testReturnTimeout := 10
+	isIgnoreUnknownFieldsEnabled := true
+	expectedHrefTest1 := "api/storage/volumes?return_records=true&fields=name,statistics,svm&return_timeout=10&ignore_unknown_fields=true"
+	hrefTest1 := NewHrefBuilder().
+		APIPath(testQuery).
+		Fields(testFields).
+		HiddenFields(testHiddenFields).
+		Filter(testFilter).
+		ReturnTimeout(&testReturnTimeout).
+		IsIgnoreUnknownFieldsEnabled(isIgnoreUnknownFieldsEnabled).
+		Build()
+
+	if hrefTest1 != expectedHrefTest1 {
+		t.Errorf("hrefTest1 should be %s but got %s", expectedHrefTest1, hrefTest1)
+	}
+
+	testFields = make([]string, 0)
+	for i := range 100000000 {
+		testFields = append(testFields, "Test"+strconv.Itoa(i))
+	}
+
+	expectedHrefTest2 := "api/storage/volumes?return_records=true&fields=*,statistics&return_timeout=10&ignore_unknown_fields=true"
+	hrefTest2 := NewHrefBuilder().
+		APIPath(testQuery).
+		Fields(testFields).
+		HiddenFields(testHiddenFields).
+		Filter(testFilter).
+		ReturnTimeout(&testReturnTimeout).
+		IsIgnoreUnknownFieldsEnabled(isIgnoreUnknownFieldsEnabled).
```

Comment:
isIgnoreUnknownFieldsEnabled is always true so this can be changed to
`IsIgnoreUnknownFieldsEnabled(true).`